contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -371,11 +393,11 @@ we define comprehensive state machines for objects, nor behaviors associated
 with state transitions. The system is level-based rather than edge-triggered,
 and should assume an Open World.
 
-An example of an oscillating condition type is `Ready` (despite it running
-afoul of current guidance), which indicates the object was believed to be fully
-operational at the time it was last probed. A possible monotonic condition
-could be `Failed`. A `True` status for `Failed` would imply failure with no
-retry. An object that was still active would generally not have a `Failed`
+An example of an oscillating condition type is `Ready`, which indicates the
+object was believed to be fully operational at the time it was last probed. A
+possible monotonic condition could be `Succeeded`. A `True` status for
+`Succeeded` would imply completion and that the resource was no longer
+active. An object that was still active would generally not have a `Succeeded`
```

Comment:
Note that `Failed` and `Succeeded` are not actually opposites and wouldn't communicate the same information. If you switched this because of the "prefer positive polarity" guidance, it's maybe a good example of something that needs an exception.

Reply:
Agree that "Failed" and "Succeeded" aren't opposites. Job is actually a great example of this -- it has two Conditions ("Complete" and "Failed"). The possible states for a Job are:

| Complete | Failed | meaning |
| --- | --- | --- |
| unset | unset | Still running |
| True | unset | Succeeded |
| True | True | Failed |

Note that many (most?) Kubernetes objects indicate a desire for an ongoing condition, rather than a one-time run, so while "Failed" may indicate having come to a flaming halt, but the opposite is probably "hasn't come to a flaming halt _yet_", i.e. "Ready", rather than "Succeeded".